File: leetcode/Algorithms/Easy/9 Palindrome Number/Solution.cpp
#include <iostream>

class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0) {
            return false;
        }

        long long num = x;
        long long reversed = 0;

        while (num > 9) {
            reversed = (reversed + num % 10) * 10;
            num /= 10;
        }
        reversed += num;

        return x == reversed;
    }
};

File: leetcode/Algorithms/Easy/9 Palindrome Number/solution.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False

        num = x
        ans = 0

        while num >= 10:
            ans = 10 * (ans + num % 10)
            num = num // 10

        ans += num

        return x == ans


num = 121

s = Solution()
print(s.isPalindrome(num))
